### DISCUSSION TITLE
QItem's "to" property will lead to a warning: <router-link>'s event prop is deprecated and has been removed in Vue Router 4...

### DISCUSSION THREAD

**Discussion by guogangj** [quasarframework/quasar#10291] [General - Components / Directives / etc] [answered]
QItem's "to" property will lead to a warning: <router-link>'s event prop is deprecated and has been removed in Vue Router 4. Use the v-slot API to remove this warning: https://next.router.vuejs.org/guide/migration/#removal-of-event-and-tag-props-in-router-link.
Quasar Version: 1.15.23
Step to produce:
1, Create a simple default Quasar App by quasar-cli
2, Replace Mainlayout.vue's left drawer's code by these content:

    <q-drawer v-model="leftDrawerOpen" show-if-above bordered content-class="bg-grey-1">
      <q-list>
        <q-item-label header class="text-grey-8">Essential Links</q-item-label>
        <q-item clickable exact exact-active-class="text-primary text-weight-bold bg-grey-4"
          :to="link.link" :key="link.title" v-for="link in essentialLinks">
          <q-item-section avatar>
            <q-icon :name="link.icon" />
          </q-item-section>
          <q-item-section>
            {{ link.title}}
          </q-item-section>
        </q-item>
      </q-list>
    </q-drawer>

3, Run this app by `quasar dev` and open browser to view the main page with DevTools then you will see the warning.

The key point is the "to" property of q-item component. No warning appears if this property is not specified.

**Comment by hawkeye64** [comment, marked as answer]
@guogangj This is just a warning and will not be displayed in your production build. Basically, it's telling you that when you migrate to vue-router v4 (vis-a-vis Quasar v2) you will need to make some changes.
